text/0000-components.md

```diff
@@ -0,0 +1,261 @@
+(when necessary, as different command-line options). 
+
+The Grid CLI will use the Rust clap library to parse command line arguments,
+with restrictions. Problematic multiple-value syntax such as `-o <val1> <val2>`
+should be avoided and `-o <val1> -o <val2>` form used instead.
+
+## SDK
+
+The Grid SDK provides programming support for application developers and
+systems integrators. Initially, only the Rust programming language is
+supported, though others may be proposed in future RFCs.
+
+In particular, the SDK includes APIs and implementations to help write smart
+contracts, including:
+
+- Reading product information from state
+- Reading Pike identity information from state
+- Reading schema information from state
+- Pike permission verifier
+- Schema compliance verify
+
+Like the other components, the SDK will grow over time as new features are
+supported by Grid. When adding new features or making changes to the SDK,
+backwards compatibility should be considered. The SDK’s versioning should
+conform to semver (which is true of all components, but of particular
+importance for the SDK).
+Smart Contracts
+
+Several features of Hyperledger Grid use smart contracts as part of their
+implementation. The smart contracts are deployed in Sabre, which is part of
+Hyperledger Sawtooth. Sabre executes smart contracts which have been compiled
+to WebAssembly. Most of the smart contracts which are formally part of
+Hyperledger Grid will be written in the Rust programming language. However,
+future RFCs may extend this to additional programming languages supported by
+future versions of Sabre.
+
+Some initial smart contracts supported by Grid are:
+
+- Grid Product
+- Grid Pike
+- Grid Track and Trace
+- Grid Schemas
+
+Smart contracts added to Grid will likely always be accompanied by an RFC which
+documents the problem they solve and technical details needed to write and
+maintain the smart contract.
+
+## Documentation
+
+Hyperledger Grid’s documentation plays a critical role in describing Grid to
+potential users and providing necessary guide and reference material for
+application developers, systems integrators, and systems administrators.
+
+The team feels strongly that “documentation is source code” and the approach to
+maintaining it matches the processes we use for all other development. As such,
+we have chosen formats for documentation which are friendly to source control
+under git while providing the best output possible. The documentation is
+written in: a) Rust doc comments for SDK API documentation; b) sphinx-doc for
+the bulk of non-API documentation; c) Markdown friendly for viewing directly in
+github for specific other documents.
+
+The sphinx-doc documentation will include an introduction to Grid, user guides,
+developer guides, system administration guides, CLI reference, smart contract
+specifications, and more.
+
+As new additions to Grid are added the documentation should be updated as well.
+
+# Reference-level explanation
```

Comment:
This title is confusing as the "name" of this particular section.  This section explains the repo/source code layout, which (I suppose) could be considered implementation information. But it doesn't have any implementation info for the preceding sections in this document.